Makefile

```diff
@@ -351,6 +351,7 @@ utils/config.ml: utils/config.mlp config/Makefile
 	    -e 's|%%WITH_PROFINFO%%|$(WITH_PROFINFO)|' \
 	    -e 's|%%WITH_SPACETIME%%|$(WITH_SPACETIME)|' \
 	    -e 's|%%WITH_SPACETIME_CALL_COUNTS%%|$(WITH_SPACETIME_CALL_COUNTS)|' \
+	    -e 's|%%FLAT_FLOAT_ARRAY%%|$(FLAT_FLOAT_ARRAY)|' \
```

Comment:
I think it might be more descriptive and less potentially confusing, for the C defines and the variable in `Config`, to just use `float_array_hack`.

Reply:
I didn't want to use "hack" because it has a negative connotation.

Reply:
I would argue that the most important thing is to have a name that really won't be confused with anything else.  There are various different concepts in the code now relating to various kinds of "float array"s and the risk of accidental confusion is I suspect quite high.  The "float array hack" may not be the ideal term, but it is the standard terminology, and so seems the right thing to me.

Reply:
@mshinwell: I do find the "hack" terminology offensive, and it is "standard" only in your mind.  Do you still want to argue pointlessly?

Reply:
100%. The only other option I could think of was float array optimization, but then you need to contract it to float_array_opt or some such thing and it can be confused with option.

Reply:
Perhaps what we want is UNTYPED_FLAT_FLOAT_ARRAY vs TYPED_FLAT_FLOAT_ARRAY (the new version).

Reply:
What about `UNBOXED_FLOAT_ARRAY`?

Reply:
(The thinking is that maybe "unboxed" is a more common term for this kind of representation than "flat".  I suppose if doing that it would make sense to rename "flat" to "unboxed" throughout the patch.)

Reply:
@xavierleroy do you agree with `unboxed` ?

Reply:
MAGICALLY_UNBOXED_FLOAT_ARRAY?  (It's not like we're getting rid of unboxed float arrays.)

Reply:
The intention of `UNBOXED_FLOAT_ARRAY` (same for `FLAT_FLOAT_ARRAY`, I imagine, too) is that it means "unboxed `float array`" (as opposed to "unboxed `floatarray`").  As such I'm not sure any extra qualification is needed.

Reply:
But they're both unboxed/flat, and that's where the confusion stems from. The difference is that one is demarcated by the type system, and one isn't.